contracts/mixins/ProtocolAccount.sol

```diff
@@ -36,13 +38,20 @@ contract ProtocolAccount is StorageLayout  {
 
     /* @notice Pays out the earned, but unclaimed protocol fees in the pool.
      * @param recv - The receiver of the protocol fees.
-     * @param token - The token address of the quote token. */
+     * @param token - The token address of the quote token.
+     * @return withdrawn tokens  */
     function disburseProtocolFees (address recv, address token) internal {
         uint128 collected = feesAccum_[token];
         feesAccum_[token] = 0;
         if (collected > 0) {
+            /**
+            * directly deposit token to fee protocol collector
+            */
             bytes32 payoutKey = keccak256(abi.encode(recv, token));
             userBals_[payoutKey].surplusCollateral_ += collected;
+            require(userBals_[payoutKey].surplusCollateral_ <= type(uint96).max, "Value exceeds uint96 range");
```

Comment:
Not really related to this change, but uint96 can theoretically overflow with something like POWA (where the total potential supply is 1 000 000 000 000 tokens with 18 decimals, or 1e30. uint96 max value is only around ~7.9e28)

Reply:
goos catch 
this is the FeeSharingCollector contract's limit.  
on the one hand - yes, it is technically possible to collect 1B, but in fact the fees are much less, practical max we had is 0.5%, so it is reduced down to 5e28 and only in case when the total supply is swapped which i doubt is practical.  
i think we should restrict tokens with more than 18 deciamls though for pools creation.